src/gluonts/support/util.py

```diff
@@ -175,3 +175,241 @@ def erfinv(x: np.array) -> np.array:
         p = c + p * w
 
     return p * x
+
+
+class Interpolation:
+    """
+    Interpolation based on trained quantile predictions
+
+    Parameters
+    ----------
+    quantile_predictions
+        Dict with key = str(quantile) and value is np.array with size prediction_length
+    interpolation_type
+
+    """
+
+    def __init__(
+        self, quantile_predictions: dict, interpolation_type: str = "linear"
+    ):
+        self.quantile_predictions = quantile_predictions
+        self.quantiles = sorted(map(float, self.quantile_predictions))
+        self.num_quantiles = len(self.quantile_predictions)
+        self.interpolation_type = interpolation_type
+
+    def __call__(self, inference_quantile):
+        if self.interpolation_type == "linear":
```

Comment:
This does not play well with extension: if you want to add new interpolation strategies, then you need to modify this class. It is better in my opinion to have a `LinearInterpolation` class with `__call__` method, and then one can add more classes with whatever interpolation

Reply:
Sure, will update the name to `LinearInterpolation`. @KelvinKan will you be adding any other additional interpolation methods here?